### PR TITLE
chore: 🤖 Add a helm chart for ethlogger

### PR DESCRIPTION
**Pull request by atoulme**
Add a Helm chart and k8s spec files for ethlogger.

Currently testing this and will add a minikube example before marking this PR for review.

### CODE REVIEW

File: deployments/k8s/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+# Source: ethlogger/templates/deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+    name: ethlogger
+    labels:
+        app.kubernetes.io/name: ethlogger
+        app.kubernetes.io/instance: ethlogger
+        app.kubernetes.io/version: '2.1.1'
```
Comment (ziegfried) [inline deployments/k8s/deployment.yaml:10]: We should probably generate this

File: deployments/k8s/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            labels:
+                app.kubernetes.io/name: ethlogger
+                app.kubernetes.io/instance: ethlogger
+        spec:
+            serviceAccountName: ethlogger
+            securityContext: {}
+            containers:
+                - name: ethlogger
+                  securityContext: {}
+                  image: 'splunkdlt/ethlogger:2.1.1'
```
Comment (ziegfried) [inline deployments/k8s/deployment.yaml:29]: We should probably generate this

File: deployments/k8s/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                        name: abis
+                  env:
+                      - name: COLLECT_BLOCKS
+                        value: 'true'
+                      - name: COLLECT_NODE_METRICS
+                        value: 'true'
+                      - name: COLLECT_NODE_INFO
+                        value: 'true'
+                      - name: COLLECT_PENDING_TX
+                        value: 'true'
```
Comment (ziegfried) [inline deployments/k8s/deployment.yaml:42]: I understand that it's useful to have all the settings here, but I'd suggest we follow the ethlogger defaults. This is disabled by default as it's really chatty and only useful in certain cases.

File: deployments/k8s/helm/ethlogger/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+version: 1.0.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application.
+appVersion: 2.1.1
```
Comment (ziegfried) [inline deployments/k8s/helm/ethlogger/Chart.yaml:21]: We should probably generate this

File: deployments/k8s/helm/ethlogger/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+# to be deployed.
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+version: 1.0.0
```
Comment (ziegfried) [inline deployments/k8s/helm/ethlogger/Chart.yaml:17]: We should probably generate this and could potentially use the ethlogger version for this

File: deployments/k8s/helm/ethlogger/values.yaml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+# Indexes to target. Omitted if set empty.
+splunkEventsIndex: "ledger_data"
+splunkMetricsIndex: "ledger_metrics"
+splunkInternalIndex: "ethlogger_internals"
+
+collectBlocks: true
+collectNodeMetrics: true
+collectNodeInfo: true
+collectPendingTransactions: true
```
Comment (ziegfried) [inline deployments/k8s/helm/ethlogger/values.yaml:40]:
```suggestion
collectPendingTransactions: false
```

File: deployments/k8s/helm/ethlogger/values.yaml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+collectPeerInfo: true
```
Comment (ziegfried) [inline deployments/k8s/helm/ethlogger/values.yaml:41]:
```suggestion
collectPeerInfo: false
```

File: deployments/k8s/serviceaccount.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+# Source: ethlogger/templates/serviceaccount.yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+    name: ethlogger
+    labels:
+        app.kubernetes.io/name: ethlogger
+        app.kubernetes.io/instance: ethlogger
+        app.kubernetes.io/version: '2.1.1'
```
Comment (ziegfried) [inline deployments/k8s/serviceaccount.yaml:10]: We should probably generate this

**Comment by atoulme** [comment]
- [x] Helm chart
- [x] k8s yaml files
- [x] helm example
- [ ] generating version numbers automatically
- [ ] regenerate yaml when helm chart changes

**Pull request merged**